File: jscomp/test/belt_result_alias_test.js
'use strict';

var Belt_Result = require("../../lib/js/belt_Result.js");

Belt_Result.map({
      TAG: /* Ok */0,
      _0: "Test"
    }, (function (r) {
        return "Value: " + r;
      }));

Belt_Result.getWithDefault(Belt_Result.map({
          TAG: /* Error */1,
          _0: "error"
        }, (function (r) {
            return "Value: " + r;
          })), "success");

/*  Not a pure module */

File: jscomp/test/belt_result_alias_test.ml
(* Make sure Belt.Result.t / Pervasives.result can be used interchangiably *)

let _ =
  Ok "Test"
  |. Belt.Result.map (fun r -> "Value: " ^ r)
  |. Js.log

let _ =
  Belt.Result.(
    Error "error"
    |. map (fun r -> "Value: " ^ r)
    |. getWithDefault("success")
  );
